### DISCUSSION TITLE
Slow NAS, fast desktop (with GPU)

### DISCUSSION THREAD

**Discussion by terryrankine** [immich-app/immich#4866] [Q&A]
NAS has slow cpu, and lots of storage
I have 400,000 photos from years of photography
I have them already in folders on my nas

was thinking of using 'external libraries'

its running in portainer on the nas

question:
how will it go..... its seems like its impossible at the moment.
can i use other PCs to do the 'processing' jobs rather than the nas? can i start a container locally and then just use whatever glue needed?

I want the library running on the NAS longterm, I have spare compute to help the digest, i have no idea the best way to tackle a big initial scan.

Its not IO limited at the moment, its CPU limited.

T

**Comment by bo0tzz** [comment]
Take a look at https://immich.app/docs/guides/machine-learning. Running microservices on a remote system is also possible, but a bit more complicated.

**Reply by jrasm91** [reply]
I'd either:
- Do the whole import on a separate machine, then migrate everything to the new machine via backup and restore. Or,
- Like bo0tzz mentioned, use external machine learning for ML at least.

**Reply by Utopiah** [reply]
> migrate everything to the new machine via backup and restore

Is this recommended for a Docker setup? Couldn't one "just" use their docker-compose file, adapt it if need be, and `scp` the data volumes?

**Comment by nodis** [comment]
I also hope to have this feature，A software called "MT photos" that can run large models on a PC, I'm not sure if immich can refer to it and implement similar functions
[https://github.com/MT-Photos/mt-photos-ai](url)

Official website：https://mtmt.tech/en/

https://github.com/MT-Photos/mt-photos-ai

Docker Hub: https://hub.docker.com/r/mtphotos/mt-photos

**Comment by Utopiah** [comment]
Check https://immich.app/docs/guides/remote-machine-learning/ that's precisely this use case.
